### PR TITLE
AUTH_LDAP_MIRROR_GROUPS ldap_config tweak

### LINKED ISSUE #445: Update AUTH_LDAP_MIRROR_GROUPS handling
## Desired Behavior

Per the Django LDAP [documentation](https://django-auth-ldap.readthedocs.io/en/latest/reference.html?highlight=AUTH_LDAP_MIRROR_GROUPS#auth-ldap-mirror-groups), the var can be set to True, False, or

> This can also be a list or other collection of group names, in which case we’ll only mirror those groups and leave the rest alone. This is ignored if `AUTH_LDAP_MIRROR_GROUPS_EXCEPT` is set.

## Contrast to Current Behavior

The `ldap_config.py` only assumes the environment variable is true or false.

## Changes Required

Update `ldap_config.py` and `docker-compose.yml` environment variables, or just document in the wiki.

## Discussion: Benefits and Drawbacks

By adding to the python, we can ensure confusion due to this current feature gap with the documented django-auth-ldap.

I can look at tackling this, but I wanted to open up for discussion and approach.  We could even look to add support for `AUTH_LDAP_MIRROR_GROUPS_EXCEPT`.

### PR DESCRIPTION
**Pull request by ryanmerolle**
Related Issue: Fixes #445

## New Behavior

If `AUTH_LDAP_MIRROR_GROUPS` not set in environment variables, it will not be set in `ldap_config.py`, allowing for ldap config shown in the new example `configuration\ldap\extra.py`

## Contrast to Current Behavior

If `AUTH_LDAP_MIRROR_GROUPS` not set in environment variables, it is set to empty in `ldap_config.py`

## Discussion: Benefits and Drawbacks

See #445

## Changes to the Wiki

TBD, likely will suggest something with regards to language around the #343 function and the example `configuration\ldap\extra.py` file included.

## Proposed Release Note Entry

- Tweaked ldap_config.py to allow for a `AUTH_LDAP_MIRROR_GROUPS` list of groups to be specified.

## Double Check

* [x] I have read the comments and followed the PR template.
* [x] I have explained my PR according to the information in the comments.
* [x] My PR targets the `develop` branch.

### CODE REVIEW

File: configuration/ldap/ldap_config.py
Comment (cimnine) [inline configuration/ldap/ldap_config.py]: I think this is not necessary, as explained here: https://github.com/netbox-community/netbox-docker/issues/445#issuecomment-784373177

File: configuration/ldap/extra.py
Comment (cimnine) [inline configuration/ldap/extra.py]: Is this here by accident or intentionally?

**Review by cimnine — DISMISSED** [review]
I like the example file!

**Review by cimnine — COMMENTED** [review]
Meant to click _Comment_, not _Approve_.

File: configuration/ldap/extra.py
Comment (ryanmerolle) [reply configuration/ldap/extra.py]: Are you referring to the whole file or the `import ldap`?

Comment (cimnine) [reply configuration/ldap/extra.py]: Just the `import ldap` line.

File: configuration/ldap/ldap_config.py
Comment (cimnine) [reply configuration/ldap/ldap_config.py]: Once https://github.com/netbox-community/netbox-docker/pull/450 is merged, I don't think this change will be necessary anymore. I would still like to keep the PR for the new `extra.py` file in the `ldap` folder.

**Comment by ryanmerolle** [comment]
I'll build from develop locally and test when I get a chance since #450 is only in develop at this point.  Give me a day or two.

**Comment by cimnine** [comment]
Do you think you have a chance to look into this anytime soon*ish? No hurry, I just want to know whether I should wait for it or otherwise postpone this change to the next release.

**Comment by ryanmerolle** [comment]
I will have this tested today or tomorrow since this directly is tied to a new deployment I have to have in place by tomorrow evening.

**Comment by ryanmerolle** [comment]
I tested as you suggested by mounting the volume to pull in the develop version of docker/configuration.docker.py from #450 

The extra.py still works as it did before.  IE if I do not comment out the variables set in ldap.py that overlap with what I declare in extra.py, the settings declared in extra.py that overlap do not take.  All the other non-overlapping variables take no matter which file they are declared in.

**Comment by ryanmerolle** [comment]
I say bump my PRs to the next release.  No need to further hold up here.

**Comment by tobiasge** [comment]
> I tested as you suggested by mounting the volume to pull in the develop version of docker/configuration.docker.py from #450
> 
> The extra.py still works as it did before. IE if I do not comment out the variables set in ldap.py that overlap with what I declare in extra.py, the settings declared in extra.py that overlap do not take. All the other non-overlapping variables take no matter which file they are declared in.

If you still have this problem you can do some more tests like this:
Mount modified versions of the configurations files into the container (with `docker-compose.override.yml`):
```
version: '3.4'
services:
  netbox:
    ports:
      - "8000:8080"
    volumes:
      - ./docker/configuration.docker.py:/opt/netbox/netbox/netbox/configuration.py:z,ro
      - ./docker/ldap_config.docker.py:/opt/netbox/netbox/netbox/ldap_config.py:z,ro
```

Modify the method `__getattr__(name)` in `docker/configuration.docker.py` and `docker/ldap_config.docker.py` to this:
```python
def __getattr__(name):
    for config in _loaded_configurations:
        try:
            attr = getattr(config, name)
            print(f"Returning '{name}' from '{config}'")
            return attr
        except:
            print(f"'{name}' not found in '{config}'")
            pass
    raise AttributeError
```

This will produce lines (a lot, basically for every settings access) like these:
```
netbox_1         | 'MAINTENANCE_MODE' not found in '<module '' from '/etc/netbox/config/extra.py'>'
netbox_1         | Returning 'MAINTENANCE_MODE' from '<module '' from '/etc/netbox/config/configuration.py'>'
```
Then you can see from where the values are loaded.

**Comment by cimnine** [comment]
@ryanmerolle , just checking what's the status here, as I lost track 🙈

**Comment by ryanmerolle** [comment]
I was quite distracted the past month with the birth of my daughter.  I plan to dig this back up along with some of my other netbox-docker items in the next week.

**Comment by ryanmerolle** [comment]
I was quite distracted the past month with the birth of my daughter.  I plan to dig this back up along with some of my other netbox-docker items in the next week.

**Comment by cimnine** [comment]
Congratulations, I hope she and your family are all well! There's no rush, just good to know that you're still on it.

**Comment by ryanmerolle** [comment]
instead of making a feature branch in my fork I developed in the develop branch.  Once this PR gets merged or rejected I will tackle #461

File: configuration/ldap/extra.py
Comment (cimnine) [reply configuration/ldap/extra.py]: I don't think you've answered my question?

File: configuration/ldap/ldap_config.py
Comment (cimnine) [reply configuration/ldap/ldap_config.py]: I believe this is not resolved either. Since #450 is now merged, the changes on these two lines should not be required any longer.

**Review by cimnine — CHANGES_REQUESTED** [review]

**Comment by ryanmerolle** [comment]
Apologies.  I forgot to commit my changes.  This should be fixed now @cimnine

**Review by cimnine — APPROVED** [review]

**Comment by cimnine** [comment]
> Apologies. I forgot to commit my changes. This should be fixed now @cimnine

Happens to all of us 😅

**Pull request merged**
